lightning/src/ln/wire.rs

```diff
@@ -248,12 +283,27 @@ pub(crate) use self::encode::Encode;
 pub trait Type: core::fmt::Debug + Writeable {
 	/// Returns the type identifying the message payload.
 	fn type_id(&self) -> u16;
+	fn is_gossip_update_msg(&self) -> bool;
```

Comment:
I was thinking you could just implement it directly here as a default implementation. BOLT 1 defines it over a message type range (256-511). That way we wouldn't accidentally forget to update the code below when a new gossip message is added.

Reply:
I think I'd kinda prefer it default to not-gossip - even if its a gossip message, if its not something we see on the wire at high volume, we may want to log messages of it at TRACE and not GOSSIP.

Reply:
In fact, I wouldn't object, though I don't feel strongly or care much, to changing the GossipTimestampFilter message to "not-gossip-update-message"

Reply:
In that case -- and given `wire` is a public module -- I think I'd prefer just adding a helper to `peer_handler` for determining what we'd like to consider gossip logging. Seems like an internal distinction that does not need to be exposed more broadly.

Reply:
Fair enough, that feels like it may be more natural.

Reply:
Alright, I reverted the implementation on `wire::Type` and introduce a simple helper function with https://github.com/rust-bitcoin/rust-lightning/pull/1145/commits/625cf8308b55fc164e8aafb55a3fc22d20f79fe7